contracts/L1/gateway/L1LPTGateway.sol

```diff
@@ -152,19 +152,19 @@ contract L1LPTGateway is IL1LPTGateway, ControlledGateway, L1ArbitrumMessenger {
         require(l1Token == l1Lpt, "TOKEN_NOT_LPT");
         (uint256 exitNum, ) = abi.decode(data, (uint256, bytes));
 
-        uint256 escrowBalance = TokenLike(l1Token).balanceOf(l1LPTEscrow);
+        uint256 escrowBalance = TokenLike(l1Lpt).balanceOf(l1LPTEscrow);
```

Comment:
I actually don't think reading the immutable var here makes any different to the end gas cost. According to [1], reading an immutable translates to a PUSH32 opcode which costs 3 gas [2]. The CALLDATALOAD opcode also costs 3 gas [2]. So, the cost for reading `l1Token` and reading `l1Lpt` should be the same. And in this case, I'd advocate for just dropping this change as it is not needed.

[1] https://blog.soliditylang.org/2020/05/13/immutable-keyword/
[2] https://github.com/crytic/evm-opcodes

Reply:
dropped the commit